packages/common/src/providers/provider.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export interface JsonRpcResponse {
+    error?: {
+        code: number;
+        message: string;
+        data: any;
+    };
+    result: any;
```

Comment:
The specification states that you only ever have `error` or `result`, but never both at the same time. We should improve the typing here.